### PR TITLE
feat(metrics): Add maximum length for metric components

### PR DESCRIPTION
**Pull request by iambriccardo**
This PR updates our develop docs with the updated limits for metric name and unit lengths.

Related to: https://github.com/getsentry/relay/pull/3628

### CODE REVIEW

File: docs/product/metrics/index.mdx [hunk 1]
```diff
@@ -47,21 +47,23 @@ Sentry incorporates protection mechanisms that will drop metrics if their cardin
...
+### Maximum Lengths
 
-Metric tags have a maximum length:
-- Tag Keys: 32 characters.
-- Tag Values: 200 characters.
+Sentry sets a maximum length for the components of a metric:
+- **Metric Name**: 150 characters.
+- **Metric Unit**: 15 characters.
+- **Metric Tag Keys**: 32 characters.
+- **Metric Tag Values**: 200 characters.
```
Comment (vivianyentran) [inline docs/product/metrics/index.mdx:66]:
```suggestion
- **Metric Tag Key**: 32 characters.
- **Metric Tag Value**: 200 characters.
```

File: docs/product/metrics/index.mdx [hunk 2]
```diff
@@ -47,21 +47,23 @@ Sentry incorporates protection mechanisms that will drop metrics if their cardin
...
 Sentry allows the use of specific characters for the components of a metric:
 
-- **Metric Name**: Supports all word characters including `_`, `-`, `.`. Any unsupported characters will be replaced by `_`.
+- **Metric Name**: All word characters including `_`, `-`, `.`. Any unsupported characters will be replaced by `_`.
 
-- **Metric Tag Keys**: Supports all word characters including `_`, `-`, `.`, `/`. Any unsupported characters will be removed from the tag key.
+- **Metric Unit**: ASCII alphanumeric characters, underscores, and digits.
 
-- **Metric Tag Values**: Supports all Unicode characters.
+- **Metric Tag Keys**: All word characters including `_`, `-`, `.`, `/`. Any unsupported characters will be removed from the tag key.
```
Comment (vivianyentran) [inline docs/product/metrics/index.mdx:54]:
```suggestion
- **Metric Tag Key**: All word characters including `_`, `-`, `.`, `/`. Any unsupported characters will be removed from the tag key.
```

File: docs/product/metrics/index.mdx [hunk 3]
```diff
@@ -47,21 +47,23 @@ Sentry incorporates protection mechanisms that will drop metrics if their cardin
...
 
-- **Custom Metric Units**:  Custom units must consist of ASCII alphanumeric characters, underscores, and digits.
+- **Metric Tag Values**: All Unicode characters.
```
Comment (vivianyentran) [inline docs/product/metrics/index.mdx:56]:
```suggestion
- **Metric Tag Value**: All Unicode characters.
```

**Review by vivianyentran — APPROVED** [review]

File: docs/product/metrics/index.mdx (hunk 1 above)
Comment (iambriccardo) [reply docs/product/metrics/index.mdx:66]: Thanks! I was torn whether to use the plural or singular, will use your suggestions

**Pull request merged**